libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs

```diff
@@ -322,6 +322,12 @@ private async Task<PromptRecognizerResult<TokenResponse>> RecognizeTokenAsync(IT
                 {
                     result.Succeeded = true;
                     result.Value = token;
+
+                    await turnContext.SendActivityAsync(new Activity { Type = ActivityTypesEx.InvokeResponse, Value = null }, cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    await turnContext.SendActivityAsync(new Activity { Type = ActivityTypesEx.InvokeResponse, Value = new InvokeResponse { Status = 500 } }, cancellationToken).ConfigureAwait(false);

```

Comment:
500 how can this be correct?
and the 500 will have the bot-quite-empty body issue right?
there is no need to say Value = null on line 326

Reply:
Removed"Value = null".

About 500, Teams hasn't got back to me about what error code they are expecting. I looked through the HTTP error codes and 500 seems the most appropriate. An unexpected error has occurred (because we expect the GetUserToken call to succeed) and no more specific message is suitable. I tested it. Teams will retry on 500 which is what we want. The body-empty issue is an issue only when we are trying to returning 200.

Reply:
For Teams to retry on a 500 is reasonable. But presumably the GetUserToken call can either _fail_ for connectivity/system reasons or decline _authorization_. Surely we should distinguish between these two situations because the first you might expect to retry and the second not. That is, the first is a 5XX (server error) and the second a 4XX (client error.)

So in our code, I would expect the first to be an exception - and therefore ultimately a 500 and the second to be a 401.